calculator.js

```diff
@@ -1,126 +1,134 @@
-function Calculator() {
-    'use strict';
-
-    let newInput = "0";
-    let oldInput = "";
-    let result = "";
-    this.input = ko.observable(newInput);
-    let operation = "";
-    let history = [];
-
-    //--------ADD NUMBER---------
-    this.addNumber = (number) => () => {
-        if (newInput === "0") {
-            newInput = number.toString();
+(function (factory) {
+    if (typeof require === 'function' && typeof exports === 'object' && typeof module === 'object') {
+        module.exports = factory(require('knockout'));
+    } else {
+        window.Calculator = factory(ko);

```

Comment:
This line of code will run if the code is running inside the browser. If that is the case, we expect that knockout is already a global variable `ko` and so we pass it into our factory function. We put the result as a global variable Calculator by doing `window.Calculator = ...`.